composer.json

```diff
@@ -24,8 +24,10 @@
     "ext-curl": "*",
     "ext-json": "*",
     "guzzlehttp/psr7": "^1.6",
-    "php-http/curl-client": "^1.7",
-    "php-http/message": "^1.7"
+    "php-http/curl-client": "^2.1",
```

Comment:
Since these required dependencies require PHP 7.1 [see here](https://github.com/php-http/curl-client/blob/2.1.0/composer.json#L20) why not bumping PHP version for the package?

Reply:
Many things could be upgraded to PHP 7.1 then like [constant visibility here](https://github.com/pay-now/paynow-php-sdk/blob/master/src/Paynow/Model/Payment/Status.php)

Reply:
Along with function/method parameter type constraints and return types.

Reply:
Looking at othe PHP libraries which use httplug they don't require specific implementation of client interface, [see Sentry for eg.](https://github.com/getsentry/sentry-php/blob/master/composer.json#L30) they only require `php-http/{discovery,httplug,client-common}` - this could relax a requirement for `"ext-curl": "*"` on platforms which are not capable of adding extensions. That was the reason HttPlug has made an abstraction around http client adapters.

---

Since you already decided to release version 2.0.0 maybe a small refactor regarding http client could be made?

Reply:
I also suggested that on my PR, then also package nyholm/psr7 could be dropped and just suggested for install by this lib

Reply:
@brzuchal Feel free to push changes

Reply:
Guys, I need this release ASAP. Let me know that you are working on any changes.

Reply:
@emilleszczak2 started yesterday and can try to finish the changes today in the evening.

Reply:
Thats not an easy task cause with Psr18Client Discovery under unit tests the mocking client is injected into HttpClient so the http client which is for made for testing purposes needs slight adjustments.

Reply:
@brzuchal Any news?

Reply:
@emilleszczak2 you can accept it or take which you wanna, I can not spend more time on this, [here's the PR](https://github.com/pay-now/paynow-php-sdk/pull/28)

Reply:
@emilleszczak2 tests should be adjusted respectively

Reply:
@brzuchal There is still a problem with mocking clients for tests. It looks like a mock-client cannot be discovered by the factory.

Reply:
> I also suggested that on my PR, then also package nyholm/psr7 could be dropped from dependencies and just suggested for install by this lib

@amodliszewski I tried to do that but I had problems with tests.

Reply:
@emilleszczak2 read the docs for *httplug* and testing there is an example of how to prepend mocking client in discovery mechanism.

Reply:
Guys tests fixed and passed.